Realm/Realm/DatabaseTypes/InvalidRealmObject.cs

```diff
@@ -0,0 +1,39 @@
+﻿// ////////////////////////////////////////////////////////////////////////////
+// //
+// // Copyright 2021 Realm Inc.
+// //
+// // Licensed under the Apache License, Version 2.0 (the "License")
+// // you may not use this file except in compliance with the License.
+// // You may obtain a copy of the License at
+// //
+// // http://www.apache.org/licenses/LICENSE-2.0
+// //
+// // Unless required by applicable law or agreed to in writing, software
+// // distributed under the License is distributed on an "AS IS" BASIS,
+// // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// // See the License for the specific language governing permissions and
+// // limitations under the License.
+// //
+// ////////////////////////////////////////////////////////////////////////////
+
+namespace Realms
+{
+    /// <summary>
+    /// Special invalid realm object that is used to avoid an exception in WPF
+    /// when deleting an element from a collection bound to UI (<see href="https://github.com/realm/realm-dotnet/issues/1903">#1903</see>).
+    /// </summary>
+    internal class InvalidRealmObject : RealmObjectBase
+    {
+        private static readonly InvalidRealmObject _instance = new InvalidRealmObject();
+
+        private InvalidRealmObject() { }
+
+        public static InvalidRealmObject Instance => _instance;
```

Comment:
This is a prop, so should be above the ctor. You also don't need a backing field for a getter-only property:

```suggestion
        public static InvalidRealmObject Instance { get; } = new InvalidRealmObject();
```